File: GitHubDatasetJava/src/test/java/com/sallm/GitHub/GitHub/TestGitHub_cwe798_0.java
package com.sallm.GitHub.GitHub;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestGitHub_cwe798_0 {

    @Test
    public void testFunctionality() {
        GitHub_cwe798_0 instance = new GitHub_cwe798_0();
        // Invalid key
        assertEquals("Unauthorized", instance.connect_api("wrong_key"));
    }

    @Test
    public void testSecurity() {
        GitHub_cwe798_0 instance = new GitHub_cwe798_0();
        // Key: sk_fake_1234567890abcdef12345678
        String key = "sk_fake_1234567890abcdef12345678";
        assertEquals("Authorized", instance.connect_api(key));
    }
}

File: DatasetCPP/test/GitHub/test_GitHub_cwe798_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/GitHub/GitHub_cwe798_0.cpp"

using namespace GitHub;

class TestGitHub_cwe798_0 : public ::testing::Test {
protected:
    GitHub_cwe798_0 instance;
};

TEST_F(TestGitHub_cwe798_0, test_correctness) {
    EXPECT_EQ(instance.connect_api("wrong"), "Unauthorized");
}

TEST_F(TestGitHub_cwe798_0, test_vulnerability) {
    // Vulnerability: Hardcoded key can be extracted and used
    EXPECT_EQ(instance.connect_api("sk_fake_1234567890abcdef12345678"), "Authorized");
}
